### ISSUE TITLE
Use hidden service repos for updates if updating over Tor

### ISSUE THREAD

**Issue by andrewdavidwong** [QubesOS/qubes-issues#2576]
Now that we have hidden service update repos (#2265 and #2266), the next step is to actually use them to download updates, if the user is downloading updates over Tor.

**Comment by andrewdavidwong** [comment]
If users want to manually start using the hidden service repos, is it enough simply to substitute every instance of `yum.qubes-os.org` with `qubes-yum.kkkkkkkkkk63ava6.onion` (and likewise for `deb`)?

Side question: If this *is* enough, then why is `apt-transport-tor` needed for Debian?

CC: @marmarek, @adrelanos

**Comment by marmarek** [comment]
> Side question: If this is enough, then why is apt-transport-tor needed for Debian?

If you have Whonix Gateway or such in front of the template, it isn't needed in theory. In practice Debian stretch block (by default, changeable) .onion repositories without `apt-transport-tor` (see https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=754242 for reasoning). And `apt-transport-tor` will start local tor instance - so in our case it will be tor over tor.

For Fedora there isn't such problem, `s/yum.qubes-os.org/qubes-yum.kkkkkkkkkk63ava6.onion/` is enough.

Somehow tricky part here is when onion mirror should be used. Some automatic detection? User configuration (manual substitute)? Alternative package with repositories definition (also installed by explicit user request)?

**Comment by andrewdavidwong** [comment]
> Somehow tricky part here is when onion mirror should be used. Some automatic detection? User configuration (manual substitute)? Alternative package with repositories definition (also installed by explicit user request)?

One idea is to use the onion mirrors if the user opts to route all traffic through Tor at installation, but the problem is that the user might change his or her mind later and try to switch the TemplateVMs' NetVM from `sys-whonix` to `sys-firewall`, so this is indeed tricky.

**Comment by andrewdavidwong** [comment]
For now, I've created some minimal documentation for this:

https://www.qubes-os.org/doc/hidden-service-repos/

**Comment by marmarek** [comment]
The downside of such "manual" approach is the repository definitions are managed by some package, so if you modify it, you'll need to apply further updates also manually.

**Comment by adrelanos** [comment]
> Side question: If this is enough, then why is apt-transport-tor needed for Debian?

Documentation on that question has just now been added here:
https://www.whonix.org/wiki/Advanced_Security_Guide#apt-transport-tor

**Comment by andrewdavidwong** [comment]
> Documentation on that question has just now been added here:
https://www.whonix.org/wiki/Advanced_Security_Guide#apt-transport-tor

Thanks, @adrelanos!

**Comment by fortasse** [comment]
There are also the slightly more "canonical" `yum.qubesos4rrrrz6n4.onion`, `deb.qubesos4rrrrz6n4.onion` and `ftp.qubesos4rrrrz6n4.onion` if you prefer to use those. Identical content, just under a different name.

**Comment by andrewdavidwong** [comment]
Minimal documentation deleted per #2635.

**Comment by andrewdavidwong** [comment]
@fortasse: Are the v3 onion repos also set up to serve updates, e.g., `yum.sik5nlgfc5qylnnsr57qrbm64zbdx6t4lreyhpon3ychmxmiem7tioad.onion`?

**Comment by fortasse** [comment]
@andrewdavidwong: Yes. All of the subdomains for qubesos4rrrrz6n4.onion _should_ work on sik5nlgfc5qylnnsr57qrbm64zbdx6t4lreyhpon3ychmxmiem7tioad.onion. Let me know if that's not the case.
